src/main/webapp/script.js

```diff
@@ -11,54 +11,113 @@ function getDashboardItems() {
         const entertainmentItemsContainer = $('#entertainmentItemsContainer');
         entertainmentItemsContainer.empty();
 
-        for (entertainmentItem of entertainmentItemsList) {
-          entertainmentItemsContainer.append(
-              createEntertainmentItemListElem(entertainmentItem));
-        }
+        populateItemGrid(entertainmentItemsContainer, entertainmentItemsList);
       })
       .catch((error) => {
-        console.log('failed to fetch entertainment items: ' + error);
+        console.log(
+            'Failed to fetch entertainment items from DashboardServlet: ' +
+            error);
       });
 }
 
 /**
- * Fetches BlobstoreServlet to get upload Url and then display submission form
- * when it becomes available.
+ * Fetches OMDb item information from OMDb API.
  */
-function getEntertainmentItemForm() {
-  fetch('/blobstore-upload?servletRedirectUrl=/item-submission')
-      .then((response) => response.text())
-      .then((imageUploadUrl) => {
-        const entertainmentItemForm = $('#entertainmentItemForm');
-        entertainmentItemForm.removeClass('hidden');
-        entertainmentItemForm.attr('action', imageUploadUrl);
+function getOMDbItem() {
+  fetch('https://www.omdbapi.com/?apikey=a28d48cd&t=' + $('#itemTitle').val())
+      .then((response) => response.json())
+      .then((OMDbItem) => {
+        const OMDbItemEntry = $('#OMDbItemEntry');
+        OMDbItemEntry.empty();
+
+        if (OMDbItem.Response === 'False') {
+          OMDbItemEntry.append($('<p>Item not found!</p>'))
+        } else {
+          OMDbItemEntry.append(createOMDbItemCard(OMDbItem));
+        }
       })
       .catch((error) => {
-        console.log('failed to fetch upload Url from Blobstore: ' + error);
+        console.log('Failed to fetch movie from OMDb API: ' + error);
       });
 }
 
 /**
- * Creates a list element that displays information about
- * a specific entertainment item.
+ * Populates the item grid used in the Dashboard with all the entertainment
+ * items on the list.
+ *
+ * @param { jQuery } entertainmentItemsContainer - the div element used as a
+ *     parent in which to add the rows and columns of the grid
+ * @param { Array } entertainmentItemsList - the list of EntertainmentItems to
+ *     add to the grid
+ */
+function populateItemGrid(entertainmentItemsContainer, entertainmentItemsList) {
+  let currItemIndex = 0;
+
+  while (currItemIndex < entertainmentItemsList.length) {
+    const rowElem = $('<div class="row mb-4"></div>');
+
+    const MAX_COLS_PER_ROW = 3;
+
+    for (let col = 1; col <= MAX_COLS_PER_ROW &&
+         currItemIndex < entertainmentItemsList.length;
+         col++, currItemIndex++) {
+      const colElem = $('<div class="col-md-4"</div>');
+      colElem.append(
+          createEntertainmentItemCard(entertainmentItemsList[currItemIndex]));
+
+      rowElem.append(colElem);
+    }
```

Comment:
it would be helpful to have some documentation for the logic here, esp the for loop that uses two variables

Reply:
I added logic documenting what is happening in the function in the last commit.